core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java

```diff
@@ -194,13 +195,19 @@ private Scanner executeFullScan(ScanAll scan, TableMetadata tableMetadata) {
     DynamoOperation dynamoOperation = new DynamoOperation(scan, tableMetadata);
     ScanRequest.Builder builder = ScanRequest.builder().tableName(dynamoOperation.getTableName());
 
-    if (scan.getLimit() > 0) {
+    // Ignore limit to control it in FilterableQueryScanner if scan has condition(s)
+    if (scan.getLimit() > 0 && scan.getConjunctions().isEmpty()) {
       builder.limit(scan.getLimit());
     }
 
-    if (!scan.getProjections().isEmpty()) {
+    List<String> projections = new ArrayList<>(scan.getProjections());
```

Comment:
Is it a bit simpler if these lines are moved to right before instantiating FilterableScanner below?

Reply:
Thanks, but we need to do this before creating `ScanRequest requestWrapper`, so it's not possible IIUC your point.

Reply:
Yeah, I overlooked `builder` was updated in this block...